Engineering/ComponentVersioning.md

```diff
@@ -0,0 +1,56 @@
+# Component Versioning
+
+## Goal
+
+Larger applications consists of multiple components that reference each other and rely on compatibility of the interfaces/contracts of the components.
+
+To achieve the goal of losely coupled applications, each component should be versioned independently hence allowing developers to detect breaking changes or seamless updates just by looking at the version number.
+
+## Version Numbers and Versioning schemes
+
+For developers or other components to detect breaking changes the version number of a component is important.
+
+There is different versioning number schemes, e.g.
+
+`major.minor[.build[.revision]]`
+
+or 
+
+`major.minor[.maintenance[.build]]`.
+
+Upon build / CI these version numbers are being generated. During CD / release components are pushed to a *component repository* such as Nuget, NPM, Docker Hub where a history of different versions is being kept.
+
+Each build the version number is incremented at the last digit.
+
+Updating the major / minor version indicates changes of the API / interfaces / contracts:
+* Major Version: A breaking change
+* Minor Version: A backwards-compatible minor change
+* Build / Revision: No API change, just a different build.
+
+## Semanting Versioning
```

Comment:
typo **Semantic** instead of **Semanting**

Reply:
True ;-)